### PR TITLE
Remove libmaus2 dependence

### PR DESCRIPTION
**Pull request by hmusta**

### CODE REVIEW

File: metagraph/src/common/serialization.hpp [hunk 1]
```diff
@@ -21,6 +31,13 @@ template <typename T>
 bool load_number_vector(std::istream &in, std::vector<T> *vector);
 
 
+void serialize_string(std::ostream &out, const std::string_view str);
```
Comment (karasikov) [inline metagraph/src/common/serialization.hpp:34]: if it's const, then better pass a const reference, there are no reasons not to.

File: metagraph/src/common/serialization.cpp [hunk 2]
```diff
@@ -329,18 +517,19 @@ bool load_number_string_map(std::istream &in, Map *map) {
         values.reserve(size);
 
         for (size_t i = 0; i < size; ++i) {
-            values.push_back(StringSerialisation::deserialiseString(in));
+            values.emplace_back();
+            load_string(in, &values.back());
```
Comment (karasikov) [inline metagraph/src/common/serialization.cpp:521]: `load_string` returns `bool`, so you have to check if it's not false.

File: metagraph/src/common/serialization.cpp [hunk 3]
```diff
@@ -350,10 +539,11 @@ bool load_number_string_map(std::istream &in, Map *map) {
...
 
         try {
-            size_t const num = NumberSerialisation::deserialiseNumber(in);
+            size_t const num = load_number(in);
             for (size_t i = 0; i < num; ++i) {
-                auto key = NumberSerialisation::deserialiseNumber32(in);
-                std::string value = StringSerialisation::deserialiseString(in);
+                auto key = load_number32(in);
+                std::string value;
+                load_string(in, &value);
```
Comment (karasikov) [inline metagraph/src/common/serialization.cpp:546]: check the return value or make it throw exceptions instead of returning `bool`

Comment (karasikov) [reply metagraph/src/common/serialization.cpp:546]: same in other similar cases...

File: metagraph/src/common/serialization.cpp [hunk 4]
```diff
@@ -106,6 +181,117 @@ template bool load_number_vector<uint8_t>(std::istream &in, std::vector<uint8_t>
 template bool load_number_vector<bool>(std::istream &in, std::vector<bool> *);
 
 
+void encode_utf8(uint32_t num, std::ostream &out) {
```
Comment (karasikov) [inline metagraph/src/common/serialization.cpp:184]: Could you add that statement "adapted from libmaus2 for backward compatibility" here as well?

**Review by karasikov — APPROVED** [review]

File: metagraph/src/common/serialization.cpp (hunk 4 above)
Comment (hmusta) [reply metagraph/src/common/serialization.cpp:184]: I ended up only keeping `decode_utf8` from libmaus2, everything else has been implemented differently.

**Pull request merged**